modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -1206,6 +1184,84 @@ impl Wallet {
             secp: &self.secp,
         }
     }
+
+    fn spawn_broadcast_pending_task(
+        task_group: &mut TaskGroup,
+        bitcoind: &DynBitcoindRpc,
+        db: Database,
+    ) {
+        task_group.spawn("broadcast pending", {
+            let bitcoind = bitcoind.clone();
+            let db = db.clone();
+            |handle| async move {
+                run_broadcast_pending_tx(db, bitcoind, &handle).await;
+            }
+        });
+    }
+
+    fn spawn_bitcoin_update_task(
+        task_group: &mut TaskGroup,
+        bitcoind: &DynBitcoindRpc,
+    ) -> (Arc<AtomicU32>, Arc<RwLock<Option<Feerate>>>) {
+        let block_count_bg = Arc::new(AtomicU32::new(0));
+        let fee_rate_bg = Arc::new(RwLock::new(None));
+
+        task_group.spawn_cancellable("wallet module: background update", {
+            let block_count_bg = block_count_bg.clone();
+            let fee_rate_bg = fee_rate_bg.clone();
+            let bitcoind = bitcoind.clone();
+            async move {
+                debug!(target: LOG_MODULE_WALLET, "Updating bitcoin block count");
+
+                let update_block_count = || async {
+                    let res = bitcoind
+                        .get_block_count()
+                        .await
+                        .and_then(|count| Ok(u32::try_from(count)?));
+
+                    match res {
+                        Ok(c) => block_count_bg.store(c, std::sync::atomic::Ordering::Relaxed),
+                        Err(err) => {
+                            warn!(target: LOG_MODULE_WALLET, %err, "Unable to get block count from the node");
+                        }
+                    }
+                };
+
+                let update_fee_rate = || async {
+                    debug!(target: LOG_MODULE_WALLET, "Updating bitcoin fee rate");
+
+                    let res = bitcoind
+                       .get_fee_rate(CONFIRMATION_TARGET).await;
+
+                    match res {
+                        Ok(Some(r)) => {
+                            *fee_rate_bg.write().await = Some(r);
+                            }
+                        Ok(None) => {
+                            debug!(target: LOG_MODULE_WALLET, "Bitcoin node did not return a fee rate");
+                        }
+                        Err(err) => {
+                            warn!(target: LOG_MODULE_WALLET, %err, "Unable to get fee rate from the node");
+                        }
+                    }
+                };
+
+                let desired_frequency = Duration::from_secs(1);
```

Comment:
Feels excessive, thinking of setups where multiple guardians poll the same `bitcoind`.

Reply:
That's the current rate of consensus proposal loop. Also 1/s is much little, so slow, very nothing.

Increasing it makes devfed starts slower, as Federation notices block height updates slower, so I was thinking about ... decreasing it... :smiling_imp:

Reply:
Actually... 

Maybe we could make it poll less often, but it needs some workaround for tests / dev env.